api/src/test/java/org/openmrs/module/locationbasedaccess/aop/EncounterSearchAdviserTest.java

```diff
@@ -0,0 +1,861 @@
+    @Test
+    public void getEncountersByPatient_nonAdminUserWithLocationPropertyShouldGetOnlyAccessibleEncounterInformation(){
+        User normalUser = userService.getUser(DEMO_USER2_ID);
+        Context.authenticate(normalUser.getUsername(), "userServiceTest");
+        assertFalse(normalUser.isSuperUser());
+
+        Location location1 = locationService.getLocation(DEMO_LOCATION1_ID);
+
+        assertEquals(normalUser.getUserProperties().size(), 1);
+        String locationUserProperty = normalUser.getUserProperty("locationUuid");
+        assertNotNull(locationUserProperty);
+        assertEquals(locationUserProperty, location1.getUuid());
+        
+        Patient patient1 = patientService.getPatient(2);
+        List<Encounter> encounterList1 = encounterService.getEncountersByPatient(patient1);
+        assertEquals(encounterList1.size(),1);
+        Encounter encounter1 = encounterList1.get(0);
+        assertNotNull(encounter1);
+        assertEquals(encounter1.getLocation(),location1);
+        
+        Patient patient2 = patientService.getPatient(7);
+        List<Encounter> encounterList2 = encounterService.getEncountersByPatient(patient2);
+        assertEquals(encounterList2.size(),2);
+        Encounter encounter2 = encounterList2.get(0);
+        Encounter encounter3 = encounterList2.get(1);
+        assertNotNull(encounter2);
+        assertEquals(encounter2.getLocation(),location1);
+        assertNotNull(encounter3);
+        assertEquals(encounter3.getLocation(),location1);
+    }
+    
+    @Test
+    public void getEncountersByVisit_adminCanAccessEncounterFromAllLocations() {
+        Context.authenticate("admin", "test");
+        User authenticatedUser = Context.getAuthenticatedUser();
+        assertNotNull(authenticatedUser);
+        assertTrue(authenticatedUser.isSuperUser());
+        
+        Location location1 = locationService.getLocation(DEMO_LOCATION1_ID);
+        Location location2 = locationService.getLocation(DEMO_LOCATION2_ID);
+        
+        Visit visit1 = visitService.getVisit(1);
+        List<Encounter> encounterList1 = encounterService.getEncountersByVisit(visit1,false);
+        assertEquals(encounterList1.size(),2);
+        Encounter encounter1 = encounterList1.get(0);
+        Encounter encounter2 = encounterList1.get(1);
+        assertNotNull(encounter1);
+        assertEquals(encounter1.getLocation(),location1);
+        assertNotNull(encounter2);
+        assertEquals(encounter2.getLocation(),location2);
+        
+        Visit visit2 = visitService.getVisit(2);
+        List<Encounter> encounterList2 = encounterService.getEncountersByVisit(visit2,false);
+        assertEquals(encounterList2.size(),1);
+        Encounter encounter3 = encounterList2.get(0);
+        assertNotNull(encounter3);
+        assertEquals(encounter3.getLocation(),location2);
+    }
+    
+    @Test
+    public void getEncountersByVisit_nonAdminUserWithOutLocationPropertyShouldNotGetEncounterInformation(){

```

Comment:
Have you checked for null user property?

Reply:
Fixed 😃